### PR TITLE
S1116 rule no-extra-semicolon

### LINKED ISSUE #497: Rule: Empty statements should be removed
Implements [RSPEC-1116](https://jira.sonarsource.com/browse/RSPEC-1116)

### PR DESCRIPTION
**Pull request by saberduck**
Fixes #497

### CODE REVIEW

File: sonarts-core/docs/rules/no-empty-statements.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# no-empty-statements
+
+Empty statements, i.e. `;`, are usually introduced by mistake, for example because:
+
+<ul>
+  <li> It was meant to be replaced by an actual statement, but this was forgotten. </li>
+  <li> There was a typo which lead the semicolon to be doubled, i.e. `;;`. </li>
+</ul>
+## See
```
Comment (vilchik-elena) [inline sonarts-core/docs/rules/no-empty-statements.md]: @saberduck for some reason this title is not displayed correctly. May be no empty line before matters

File: sonarts-core/src/rules/noEmptyStatementsRule.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+export class Rule extends tslint.Rules.AbstractRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "no-empty-statements",
+    description: "Empty statements should be removed",
+    rationale: tslint.Utils.dedent``,
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-1116",
+    type: "functionality",
```
Comment (vilchik-elena) [inline sonarts-core/src/rules/noEmptyStatementsRule.ts]: @saberduck I don't think it's functionality, but maintainability

File: sonarts-core/docs/rules/no-empty-statements.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# no-empty-statements
+
+Empty statements, i.e. `;`, are usually introduced by mistake, for example because:
+
+<ul>
+  <li> It was meant to be replaced by an actual statement, but this was forgotten. </li>
+  <li> There was a typo which lead the semicolon to be doubled, i.e. `;;`. </li>
```
Comment (vilchik-elena) [inline sonarts-core/docs/rules/no-empty-statements.md]: @saberduck update RSPEC for TS coping JS (title + 3rd point in this list + examples)

File: sonarts-core/src/rules/noEmptyStatementsRule.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    typescriptOnly: false,
+  };
+
+  public static MESSAGE = "Remove this empty statement.";
```
Comment (vilchik-elena) [inline sonarts-core/src/rules/noEmptyStatementsRule.ts]: @saberduck make message as in JS (`Remove this extra semicolon.`). It's more explicit

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/resources/org/sonar/l10n/typescript/rules/typescript/S1116.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<p>Empty statements, i.e. <code>;</code>, are usually introduced by mistake, for example because:</p>
```
Comment (vilchik-elena) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/resources/org/sonar/l10n/typescript/rules/typescript/S1116.html]: @saberduck regenerate metadata after update (JS -> TS)

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/resources/org/sonar/l10n/typescript/rules/typescript/S1116.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (vilchik-elena) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/resources/org/sonar/l10n/typescript/rules/typescript/S1116.json]: @saberduck regenerate

**Comment by vilchik-elena** [comment]
Tested manually

**Comment by saberduck** [comment]
@vilchik-elena please have another look

**Comment by vilchik-elena** [comment]
@saberduck 👍 merge when green

**Pull request merged**